### PR TITLE
fix -longtics parameter, remove unnecessary variables

### PR DESCRIPTION
**Pull request by rfomin**
Fixes this issue: [[doomworld]](https://www.doomworld.com/forum/topic/112333-this-is-woof-810-nov-26-2021-updated-winmbf/?do=findComment&comment=2427340)

### CODE REVIEW

**Comment by rfomin** [comment]
https://github.com/kraflab/dsda-doom/blob/e0f2300a465295a617350f0eda8ebf3af819188f/prboom2/src/g_game.c#L3550
@kraflab does this mean that DSDA-Doom runs in -shorttics mode by default?
Not sure if we need this in a more casual Woof.

**Comment by fabiangreffrath** [comment]
AFAIUI, regular game play should always be with long tics. Demo recording should always be with short tics, unless `-longtics` is given and complevel is Vanilla, or complevel is MBF21.

File: Source/g_game.c [hunk 1]
```diff
@@ -123,7 +123,6 @@ boolean         haswolflevels = false;// jff 4/18/98 wolf levels present
 byte            *savebuffer;
 int             autorun = false;      // always running?          // phares
 
-static int      complevel = MBFVERSION;
```
Comment (fabiangreffrath) [inline Source/g_game.c:126]: This looks like a lot of unrelated changes. Is this necessary?

Comment (rfomin) [reply Source/g_game.c:126]: Not required, but this additional `complevel` variable is unnecessary. This code is quite convoluted, so I want to simplify it a little.

**Comment by kraflab** [comment]
> https://github.com/kraflab/dsda-doom/blob/e0f2300a465295a617350f0eda8ebf3af819188f/prboom2/src/g_game.c#L3550 @kraflab does this mean that DSDA-Doom runs in -shorttics mode by default? Not sure if we need this in a more casual Woof.

Yes, when demo recording it defaults to shorttics mode.

File: Source/g_game.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/g_game.c:126]: Okay, fine! 👍

**Comment by fabiangreffrath** [comment]
> Yes, when demo recording it defaults to shorttics mode.

Even in complevel MBF21?

**Comment by kraflab** [comment]
> Even in complevel MBF21?

Yes

**Comment by fabiangreffrath** [comment]
Should we follow suit?

**Comment by rfomin** [comment]
I suggest leaving complevel MBF21 only in `longtics` mode and waiting until someone complains. Users can also add the `-shorttics` parameter.

**Comment by fabiangreffrath** [comment]
> Yes

Was this an oversight or a deliberate decision? MBF21 could have been the opportunity to change this default.




> I suggest leaving complevel MBF21 only in `longtics` mode and waiting until someone complains. Users can also add the `-shorttics` parameter.

One could also do the inverse with the same argumentation. 😉

**Comment by rfomin** [comment]
> Was this an oversight or a deliberate decision? MBF21 could have been the opportunity to change this default.

I think the community believes that some of the tricks are easier in `-shorttics` mode or they just got used to it. PrBoom+ demos (demo_version == 214) are in `longtics` mode by default, but they are not popular.

**Comment by kraflab** [comment]
The demos for mbf21 always store with longtics precision, so its up to the port's discretion when to truncate with shorttics. Since dsda-doom is for speedrunning primarily, and most precise speedrunning tricks require shorttics, this makes sense as the default there. I also don't see any benefit to changing this behaviour when it's the same for all other relevant complevels.
